components/brave_wallet/browser/brave_wallet_controller.cc

```diff
@@ -0,0 +1,229 @@
+std::string BraveWalletController::GetRandomNonce() {
+  // crypto::RandBytes is fail safe.
+  uint8_t nonceBytes[kNonceByteLength];
+  crypto::RandBytes(nonceBytes, kNonceByteLength);
+  return std::string(
+      reinterpret_cast<char*>(nonceBytes), kNonceByteLength);
+}
+
+// static
+std::string BraveWalletController::GetRandomSeed() {
+  // crypto::RandBytes is fail safe.
+  uint8_t random_seed_bytes[kSeedByteLength];
+  crypto::RandBytes(random_seed_bytes, kSeedByteLength);
+  return std::string(
+      reinterpret_cast<char*>(random_seed_bytes), kSeedByteLength);
+}
+
+// static
+bool BraveWalletController::SealSeed(const std::string& seed,
+    const std::string& key, const std::string& nonce,
+    std::string* cipher_seed) {
+  crypto::Aead aes_256_gcm_siv(crypto::Aead::AES_256_GCM_SIV);
+  aes_256_gcm_siv.Init(&key);
+  return aes_256_gcm_siv.Seal(base::StringPiece(seed.begin(), seed.end()),
+      nonce, base::StringPiece(""), cipher_seed);
+}
+
+// Store the seed in preferences, binary pref strings need to be
+// base64 encoded.  Base64 encoding is fail safe.
+// static
+void BraveWalletController::SaveToPrefs(
+    Profile* profile,
+    const std::string& cipher_seed,
+    const std::string& nonce) {
+  // Store the seed in preferences, binary pref strings need to be
+  // base64 encoded.  Base64 encoding is fail safe.
+  std::string base64_nonce;
+  std::string base64_cipher_seed;
+  base::Base64Encode(nonce, &base64_nonce);
+  base::Base64Encode(base::StringPiece(cipher_seed.begin(),
+                                      cipher_seed.end()),
+      &base64_cipher_seed);
+  profile->GetPrefs()->SetString(kBraveWalletAES256GCMSivNonce, base64_nonce);
+  profile->GetPrefs()->SetString(kBraveWalletEncryptedSeed,
+      base64_cipher_seed);
+}
+
+void BraveWalletController::ResetCryptoWallets() {
+  Profile* profile = Profile::FromBrowserContext(context_);
+  const base::FilePath wallet_data_path = profile->GetPath()
+      .AppendASCII("Local Extension Settings")
+      .AppendASCII(ethereum_remote_client_extension_id);
+
+  base::PostTaskAndReplyWithResult(file_task_runner_.get(), FROM_HERE,
+      base::BindOnce(&ResetCryptoWalletsOnFileTaskRunner, wallet_data_path),
+      base::BindOnce(&BraveWalletController::OnCryptoWalletsReset,
+                     weak_factory_.GetWeakPtr()));
+}
+
+void BraveWalletController::OnCryptoWalletsReset(bool success) {
+  if (success) {
+    BraveWalletController::CloseTabsAndRestart();
+  }
+}
+
+void BraveWalletController::CloseTabsAndRestart() {
+  // Close all CW tabs in each browser instance
+  for (auto* browser : *BrowserList::GetInstance()) {
+    auto* tab_strip = browser->tab_strip_model();
+    for (int i = 0; i < tab_strip->count(); ++i) {
+      auto* web_contents = tab_strip->GetWebContentsAt(i);
+      GURL url = web_contents->GetURL();
+      if (url.SchemeIs(content::kChromeUIScheme) &&
+          url.host() == ethereum_remote_client_host) {
+        web_contents->Close();
```

Comment:
Does this actually get rid of the tab_strip's webcontents too? if so this might cause some wonkiness with the iterator on the count.

Reply:
I'll do some tests and double check, I'll report back 👍